### PR TITLE
TS-4550 Adds X-Transaction-Id to xdebug

### PR DESCRIPTION
**Pull request by zwoop**
This makes a header value that matches the new log tag, %<cruuid>.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/354/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/248/ for details.

File: plugins/experimental/xdebug/xdebug.cc [hunk 1]
```diff
@@ -234,6 +237,20 @@ InjectMilestonesHeader(TSHttpTxn txn, TSMBuffer buffer, TSMLoc hdr)
...
+{
+  TSMLoc dst = FindOrMakeHdrField(buffer, hdr, "X-Transaction-Id", lengthof("X-Transaction-Id"));
+
+  if (TS_NULL_MLOC != dst) {
+    char buf[TS_UUID_STRING_LEN + 22]; // Padded for int64_t (20) + 1 ('-') + 1 ('\0')
+    TSUuid uuid = TSProcessUuidGet();
+    int len     = snprintf(buf, sizeof(buf) - 1, "%s-%" PRIu64 "", TSUuidStringGet(uuid), TSHttpTxnIdGet(txn));
+
+    TSReleaseAssert(TSMimeHdrFieldValueStringInsert(buffer, hdr, dst, 0 /* idx */, buf, len) == TS_SUCCESS);
+  }
```
Comment (jpeach) [inline plugins/experimental/xdebug/xdebug.cc]:
```C
  if (dst != TS_NULL_MLOC) {
    TSHandleMLocRelease(buffer, hdr, dst);
  }
```

Comment (jpeach) [inline plugins/experimental/xdebug/xdebug.cc]: Actually ``TSHandleMLocRelease`` accepts ``TS_NULL_MLOC`` as the last argument, so all the places here can do away with the null check.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/384/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/280/ for details.

**Pull request closed without merging**